tests/cpp/test_reduction_pointwise.cpp

```diff
@@ -0,0 +1,122 @@
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <ops/all_ops.h>
+#include <scheduler/all_schedulers.h>
+#include <scheduler/reduction_utils.h>
+#include <scheduler/utils.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+namespace nvfuser {
+
+using PointwiseFusedReductionTest = NVFuserTest;
+
+// inner reduction + non-broadcast epilogue, can't be fused
+// outer reduction + non-broadcast epilogue, can be fused
+// checked by: SchedulerTopologyChecker::supportedPostReductionFusion
+namespace {
+void ReductionNonBroadcast(const int reduction_dim) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  const int dim0 = 8192;
+  const int dim1 = 1024;
+  DataType input_dtype = DataType::Float;
+  auto tv0 = makeContigTensor(2, input_dtype);
+  auto tv1 = makeContigTensor(1, input_dtype);
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  auto tv2 = sum(tv0, {reduction_dim});
+  auto tv3 = add(tv2, tv1);
+  fusion->addOutput(tv3);
+
+  auto options = at::TensorOptions()
+                     .dtype(data_type_to_aten(input_dtype))
+                     .device(at::kCUDA, 0);
+  auto t0 = at::randn({dim0, dim1}, options);
+  auto t1 = at::randn({reduction_dim == 0 ? dim1 : dim0}, options);
+
+  // check whether the fusion can be scheduled as reduction
+  SchedulerRuntimeInfo runtime_info(fusion.get(), {t0, t1});
+  bool can_schedule = SchedulerEntry::canSchedule(
+      ScheduleHeuristic::Reduction, fusion.get(), runtime_info);
+  if (reduction_dim == 1) {
+    ASSERT_FALSE(can_schedule);
+  } else {
+    ASSERT_TRUE(can_schedule);
+  }
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  auto cg_outputs = executor_cache.runFusionWithInputs({t0, t1});
+
+  auto t_ref = t0.sum({reduction_dim}) + t1;
+  testValidate(
+      executor_cache.fusion(),
+      cg_outputs,
+      {t0, t1},
+      {t_ref},
+      __LINE__,
+      __FILE__);
+};
+} // namespace
+TEST_F(PointwiseFusedReductionTest, InnerReductionNonBroadcast) {
+  const int reduction_dim = 1;
+  ReductionNonBroadcast(reduction_dim);
+}
+TEST_F(PointwiseFusedReductionTest, OuterReductionNonBroadcast) {
```

Comment:
so even though we are adding all four tests, the `OuterReductionNonBroadcast` is really the only one that the entire fusion goes through a single kernel. The other three would just segment into `reduction` + `pointwise`?

Reply:
You are right. I believe all other three cases are segmented due to performance reason and we need to revisit this decision by optimizing the generated code and add corresponding benchmark cases in the followed PRs.